### PR TITLE
Switch uuid library to gofrs/uuid

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/azure-sdk-for-go/issues/3158

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
You will need to run `gulp updatetestdep` to update the test dependencies with the new package.

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT I just updated the dependencies. But please hold on merging, I am working to see how many packages are impacted and if not so many, we could regenerate all of them in the next major version.

I am opening a PR in go SDK to see whether the APIs in these two packages are identical

File: test/src/tests/Gopkg.lock [hunk 1]
```diff
@@ -27,12 +27,12 @@
   version = "v3.2.0"
 
 [[projects]]
-  digest = "1:4cc7748b74453830951f95ddc3fd980387e7c13f77cc33d7cba54cd6b96dad26"
-  name = "github.com/kr/pretty"
+  digest = "1:73770491417b7cb646b28610eec6481187aba7b5c3703da0d42f8d590d016f5f"
+  name = "github.com/gofrs/uuid"
```
Comment (jhendrixMSFT) [inline test/src/tests/Gopkg.lock:31]: Hmm looks like `gopkg.toml` wasn't updated.  You might need to update it by hand and rerun `gulp updatetestdep`

Comment (ArcturusZhang) [reply test/src/tests/Gopkg.lock:31]: I have updated the toml, please have a look, thanks

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@ArcturusZhang when do you want to release with this change?  Also note the merge conflicts.

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang when do you want to release with this change? Also note the merge conflicts.

it may be better if we only have one big series of breaking changes in one major version.
Considering we just made the future mocking change, could we postpone this to the next next major version after we regenerate everything to pick up the future mocking fix?

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT could you please merge this? We would like to release this change in this week's release.

**Comment by Porges** [comment]
Thanks!

**Pull request merged**